cou/steps/plan.py

```diff
@@ -349,15 +349,61 @@ def generate_common_plan(
                 raise_on_blocked=True,
             ),
         )
-    )
+    ]
     if args.backup:
-        plan.add_step(
+        steps.append(
             PreUpgradeStep(
                 description="Backup mysql databases",
                 coro=backup(analysis_result.model),
             )
         )
-    return plan
+
+    return steps
+
+
+def _get_post_upgrade_steps(analysis_result: Analysis, args: CLIargs) -> list[PostUpgradeStep]:
+    """Get the post-upgrade steps.
+
+    :param analysis_result: Analysis result
+    :type analysis_result: Analysis
+    :param args: CLI arguments
+    :type args: CLIargs
+    :return: List of post-upgrade steps.
+    :rtype: list[PreUpgradeStep]
+    """
+    steps = []
+    if args.upgrade_group in {DATA_PLANE, None}:
+        steps.append(_get_ceph_mon_post_upgrade_step(analysis_result))
+
+    return steps
+
+
+def _get_ceph_mon_post_upgrade_step(analysis_result: Analysis) -> PostUpgradeStep:
+    """Get the post-upgrade step for ceph-mon, where we check the require-osd-release option.
+
+    :param analysis_result: Analysis result
+    :type analysis_result: Analysis
+    :return: The post-upgrade step.
+    :rtype: PreUpgradeStep
+    :raises NotSupported: When two ceph-mon are found at the model.
+    """
+    ceph_mons_apps = [
+        app for app in analysis_result.apps_control_plane if isinstance(app, CephMon)
+    ]
+    match num_ceph_mon_apps := len(ceph_mons_apps):
+        case 0:
+            return PostUpgradeStep()
+        case 1:
+            app = ceph_mons_apps[0]  # getting first and only one ceph-mon app
+            unit = list(app.units.values())[0]  # getting first unit, sice we do not care which one
+            return PostUpgradeStep(
+                "Ensure require-osd-release option matches with ceph-osd version",
+                coro=set_require_osd_release_option(unit.name, app.model),
```

Comment:
As we discussed on COU sync, we can `set_require_osd_release_option` for all ceph-mon applications